src/obstacle_detector.cpp

```diff
@@ -158,14 +165,70 @@ void ObstacleDetector::sensor_callback(const sensor_msgs::Range::ConstPtr &msg)
     else {
         RangeSensor& sensor = sensors[frame];
         sensor.update(msg->range, msg->header.stamp);
-        draw_line(sensor.origin, sensor.left_vertex, 0, 0, 1, sensor.id + 200);
-        draw_line(sensor.origin, sensor.right_vertex, 0, 1, 0, sensor.id + 300);
-        draw_line(sensor.left_vertex, sensor.right_vertex, 0.5, 0.5, 0,
+        draw_line(sensor.origin, sensor.left_vertex, 0.5, 0.5, 0.5, sensor.id + 200);
+        draw_line(sensor.origin, sensor.right_vertex, 0.5, 0.5, 0.5, sensor.id + 300);
+        draw_line(sensor.left_vertex, sensor.right_vertex, 1, 1, 1,
             sensor.id + 400);
     }
     obstacle_mutex.unlock();
 }
 
+void ObstacleDetector::scan_callback(const sensor_msgs::LaserScan::ConstPtr &msg)
+{
+    float theta = msg->angle_min;
+    float increment = msg->angle_increment;
+    float range_min = msg->range_min;
+
+    if (!have_lidar) {
+        try {
+            geometry_msgs::TransformStamped tfs =
+                tf_buffer->lookupTransform("base_link", "laser", ros::Time(0));
```

Comment:
A more sensical name for the variable like `laser_tf` or `base2laser_tf` would make the following code clearer